external/solr/setup-solr.sh

```diff
@@ -0,0 +1,59 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#!/bin/bash
+
+STATUS_SHARDS=$(grep 'solr.status.routing.shards' solr-conf.yaml | sed -e 's/.*: //' | tr -d ' ')
+ROUTER_FIELD=$(grep 'solr.status.routing.fieldname' solr-conf.yaml | sed -e 's/.*: //' | tr -d ' ')
+
+SOLR_PORT=8983
+SOLR_HOME=/opt/solr-9.7.0
+
+$SOLR_HOME/bin/solr start -cloud -p $SOLR_PORT
+
+$SOLR_HOME/bin/solr zk upconfig -n "docs" -d configsets/docs -z localhost:9983
+$SOLR_HOME/bin/solr zk upconfig -n "status" -d configsets/status -z localhost:9983
+$SOLR_HOME/bin/solr zk upconfig -n "metrics" -d configsets/metrics -z localhost:9983
+
+echo "Creating 'docs' collection ..."
+curl -X POST "http://localhost:$SOLR_PORT/api/collections" -H "Content-type:application/json" -d '
+  {
+    "name": "docs",
```

Comment:
does this replace the core.properties files?

Reply:
Yes. Since we are switching to SolrCloud, the recommended way to create a collection is through the [Collections API](https://solr.apache.org/guide/solr/latest/deployment-guide/collection-management.html#create), where `name` is a required parameter. Additionally, in SolrCloud the configuration (configsets) and data are [managed by ZooKeeper](https://solr.apache.org/guide/solr/latest/configuration-guide/config-sets.html#configsets-in-solrcloud-clusters), so my understanding is that `core.properties` is not relevant anymore.